File: 172.cpp
class Solution {
public:
    /** 
     *@param A: A list of integers
     *@param elem: An integer
     *@return: The new length after remove
     */
    int removeElement(vector<int> &A, int elem) {
        // write your code here
    	int l=A.size();
        int total=l;

        A.erase(remove(A.begin(),A.end(),elem),A.end());
        return A.size();

        // vector<int> B=
        // for (vector<int>::iterator s=A.begin();s!=A.end();++s){
        // 	if ( *s==elem){
        // 		total -= 1;
        // 		A.erase(s);
        // 	}
        // }


    }
};

File: 172.py
class Solution:
    """
    @param A: A list of integers
    @param elem: An integer
    @return: The new length after remove
    """
    def removeElement(self, A, elem):
        # write your code here

        #A=[el for el in A if el != elem]
        
        self.A=filter(lambda x: x != elem,A)
        A=self.A
        # while elem in A:
        # 	del elem

        return len(self.A)

